connect/src/com/telenor/connect/SdkProfile.java

```diff
@@ -24,10 +24,23 @@
     WellKnownAPI.WellKnownConfig getWellKnownConfig();
     boolean isInitialized();
 
-    void onStartAuthorization(OnStartAuthorizationCallback callback);
+    void initializeFromUri(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            Uri initFrom,
+            OnDeliverAuthorizationUriCallback callback);
 
-    interface OnStartAuthorizationCallback {
-        void onSuccess();
+    void onStartAuthorization(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            OnStartAuthorizationCallback callback);
+
+    interface OnDeliverAuthorizationUriCallback {
+        void onSuccess(Uri authorizationStartUri);
         void onError();
     }
+
+    interface OnStartAuthorizationCallback extends OnDeliverAuthorizationUriCallback {
```

Comment:
I kind of disagree for a couple of reasons listed below (ideally, operator selection should have been a part of a normal OAuth2 flow, in which case we wouldn't have had discussions of this kind in the first place). 

1. A separate OperatorSelection activity would require publishing it in the manifest, thus exposing developers to an activity they are not supposed to use. I think they generally benefit from encapsulations of this kind (the less - the better).

2. ConnectWebViewClient is not modified with OperatorSelection code, but with a more generic thing that the MobileConnect uses for operator selection. Connect Id could have used it for something else, provided they decided to implement preamble diversions. The fact that they don't do it now does not mean they couldn't.